src/web/browser/App.tsx

```diff
@@ -143,16 +139,20 @@ const App = ({ CAPI, NAV }: Props) => {
             <Portal root="most-viewed-right">
                 <MostViewedRightWrapper pillar={CAPI.pillar} />
             </Portal>
-            <Portal root="slot-body-end">
-                <SlotBodyEnd
-                    contentType={CAPI.contentType}
-                    sectionName={CAPI.sectionName}
-                    shouldHideReaderRevenue={CAPI.shouldHideReaderRevenue}
-                    isMinuteArticle={CAPI.pageType.isMinuteArticle}
-                    isPaidContent={CAPI.pageType.isPaidContent}
-                    tags={CAPI.tags}
-                />
-            </Portal>
+            {isSignedIn !== undefined && countryCode !== undefined && (
```

Comment:
Only render the slot portal when we've determined if the user is logged in as well as their geolocation.

Reply:
Why are login status and country code presence gates for whether to render the slot?

Reply:
```suggestion
            {isSignedIn && countryCode  && (
```
do we need to check if `undefined`? cant we check if truthy?

Reply:
Nope, we need `undefined` specifically as either of those vars can be `true` or `false` - `undefined` means we haven't determined them yet

Reply:
> Why are login status and country code presence gates for whether to render the slot?

Because the component needs these 2 bits of data to have been defined before it can fetch the remote Epic. The values can be `true` or `false`, but they must be defined (in this context, being defined means exactly that - we've checked and the user is either logged in or not logged in)

Reply:
Ah thanks for the explanation. Thoughts on extracting to a function (or pair of functions) in order to give this a clear name? Something like `hasDeterminedLoggedInStatus && hasDeterminedCountryCode`?

Reply:
@andre1050 and I discussed this yesterday. The component needs to wait for both values to have been _decided_ so rather than use a separate `loginStatusDecided` prop it seemed more semantic to check if the prop was `undefined`

Reply:
Ah okay. As long as those values are guaranteed to resolve I think this is fine. The slot shouldn't be conditional on data that may not exist though if that makes sense?

Reply:
There's the potential that an api call will fail or an attempt to read from localStorage fails - it would be good to step through these scenarios and make sure we're returning sensible defaults.

For `isSignedIn` we can default this to `false` on error.
For `countryCode`, ? @nicl should/can the slot make the api call if it doesn't know the country code?

Reply:
Thanks @oliverlloyd for pairing with me on this - we've ensured neither `isSignedIn` or `countryCode` remains `undefined` so there should be no scenario where the  slot doesn't render because the gate's won't open (stretching the metaphor now, I know).

If the geolocation call fails or we can't parse the response, we'll return an empty string letting the slot render and passing that value accordingly to the service (which knows already how to handle that).